File: app/src/main/java/com/ocse/androidbaselib/retrofit/ApiService.java
package com.ocse.androidbaselib.retrofit;


import com.ocse.androidbaselib.bean.UserBean;
import com.ocse.androidbaselib.bean.VersionBean;

import java.util.Map;

import io.reactivex.Observable;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;


/**
 * @author hujiayi
 */
public interface ApiService {
    /**
     * 登录
     * @param map
     * @return
     */
    @FormUrlEncoded
    @POST("oauth/token")
    Observable<UserBean> login(@FieldMap Map<String, Object> map);


    /**
     * 登录
     * @param map
     * @return
     */
    @FormUrlEncoded
    @POST("mobile/getAppVersion")
    Observable<VersionBean> getAppVersion(@FieldMap Map<String, Object> map);
}

File: app/src/main/java/com/ocse/androidbaselib/retrofit/ApiService.kt
package com.ocse.androidbaselib.retrofit

import com.ocse.androidbaselib.bean.UserBean
import com.ocse.androidbaselib.bean.VersionBean
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.FieldMap
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

/**
 * @author hujiayi
 */
interface ApiService {
    /**
     * 登录
     * @param map
     * @return
     */
    @FormUrlEncoded
    @POST("oauth/token")
    fun login(@FieldMap map: Map<String, @JvmSuppressWildcards Any>): Observable<UserBean>

    /**
     * 登录
     * @param map
     * @return
     */
    @FormUrlEncoded
    @POST("mobile/getAppVersion")
    fun getAppVersion(@FieldMap map: Map<String, @JvmSuppressWildcards Any>): Observable<VersionBean>
}